vortex-ipc/src/io/read.rs

```diff
@@ -12,7 +12,7 @@ pub trait VortexRead {
 
 pub trait VortexReadAt {
     fn read_at_into(
-        &mut self,
+        &self,
```

Comment:
this should never be mut, if you're reading at a location you shouldn't need to mutate internal state

Reply:
i could imagine an implementation of this on top of a file, in which read_at_into would be implemented as

1. seek
2. read

are you saying we wouldn't want to implement this trait for file IO?

Reply:
Every reasonable OS and framework supports read_at with &self (not &mut self). I had implementation like this before and thought it's very silly if we ultimately know the exact size and offset we want to read

Reply:
Fwiw only tokio does a silly thing where you have to seek and read and I had an impl of that. However, it's A LOT simpler to think of ReadAts as read factories that are immutable

Reply:
hmm actually i'm a bit confused, i see in the file below there are a set of methods that don't require mut self that i had assumed would

Reply:
https://github.com/rust-lang/rust/blob/master/library/std/src/sys/pal/unix/fd.rs#L123-L146

very interesting, was not familiar with pread syscall